pkg/sql/app_stats.go

```diff
@@ -46,14 +47,16 @@ type stmtStats struct {
 
 // StmtStatsEnable determines whether to collect per-statement
 // statistics.
-var StmtStatsEnable = envutil.EnvOrDefaultBool(
-	"COCKROACH_SQL_STMT_STATS_ENABLE", true,
+var StmtStatsEnable = settings.RegisterBoolSetting(
+	"sql.metrics.stmtstats.enabled", "collect per-statement query statistics ", true,
```

Comment:
I'd spell `statement` in full, and put a separator between it and `stats`. Putting `metrics` and `stats` in the same name seems redundant.